### DISCUSSION TITLE
How to extend or modify the WS Engine

### DISCUSSION THREAD

**Discussion by code-leen** [artilleryio/artillery#1427] [Artillery]
How would I go about extending or modifying the `getMessageHandler` of Artillery's WebSocket Engine?
I want to capture and process a WebSocket's response to confirm a valid response.

**Comment by GeekyDeaks** [comment]
There is actually a non-documented `match` in the ws engine (and I think the others), so you can do something like this:

      - send:
          payload:
            type: handshake
          capture: []
          match:
            json: "$.user_id"
            value: test001

You have to provide an empty `capture` to make it work as the code doesn't currently check for `match`.   I started a PR to allow for this though #1422

**Reply by hassy** [reply]
hi @code-leen 👋 does the change in @GeekyDeaks's PR (#1422) work for your use case?

**Reply by code-leen** [reply]
Hello 👋🏼  This does seem like it would work! 🎉 
I have one more question. I've tried to find it in the documentation, but what does the `$.` prefix denote?

**Reply by GeekyDeaks** [reply]
hi @code-leen - it's jsonpath syntax: https://github.com/dchester/jsonpath

**Reply by code-leen** [reply]
Thank you so much! 🙌🏼
